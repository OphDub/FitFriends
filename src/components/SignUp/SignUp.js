import React, { Component } from 'react';
import './SignUp.css';
import { connect } from 'react-redux';
import { signUpUser, loginUser } from '../../actions/actionsIndex';
class SignUp extends Component {
  constructor () {
    super()
    this.state={
      firstName: '',
      lastName: '',
      userName: '',
      userEmail: '',
      userPass: '',
      userId: '',
    }
  }

  handleChange = (e) => {
    const { name, value } = e.target;
    this.setState({ [name]: value });
  }

  handleSignUp = (e) => {
    e.preventDefault();

    const userId = Date.now();
    this.setState({ userId });
    this.props.signUpUser(this.state);
    this.props.loginUser({user: this.state.userName});

    this.setState({
      firstName: '',
      lastName: '',
      userName: '',
      userEmail: '',
      userPass: '',
      userId: '',
    });
  }

  authWithFacebook = (e) => {
    e.preventDefault();

  }

  render () {
    return (
      <form className="SignUp">
        <div>
          <button onClick={this.authWithFacebook}>Login with Facebook</button>
        </div>
        <div className="signup-email">
          <input type="text"
            placeholder="First Name"
            name="firstName"
            value={this.state.firstName}
            onChange={this.handleChange}/>
          <input type="text"
            placeholder="Last Name"
            name="lastName"
            value={this.state.lastName}
            onChange={this.handleChange}/>
          <input type="text"
            placeholder="Username"
            name="userName"
            value={this.state.userName}
            onChange={this.handleChange}/>
          <input type="email"
            placeholder="Email"
            name="userEmail"
            value={this.state.userEmail}
            onChange={this.handleChange}/>
          <input type="password"
            placeholder="Password"
            name="userPass"
            value={this.state.userPass}
            onChange={this.handleChange}/>
          <button
            onClick={this.handleSignUp}>
              Sign Up
          </button>
        </div>
      </form>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  signUpUser: (user) => dispatch(signUpUser(user)),
  loginUser: (user) => dispatch(loginUser(user))
});

export default connect(null, mapDispatchToProps)(SignUp);