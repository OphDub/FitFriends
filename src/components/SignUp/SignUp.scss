.SignUp {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100%;
  background: linear-gradient(
    rgba(0,0,0,0.45),
    rgba(0,0,0,0.45)
  ), url('https://static.pexels.com/photos/685531/pexels-photo-685531.jpeg');
  background-repeat: no-repeat;
}

.form-background {
  background-color: #D8D8D8;
  padding: 15px;
  border-radius: 10px;
}

.SignUp-form {
  display: flex;
  flex-direction: column;
}

.SignUp-form label {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding-bottom: 10px;
}

.SignUp-form input {
  margin-left: 10px;
  padding: 5px;
  border-radius: 5px;
  border: none;
  outline: none;
}

.signup-btn {
  border: none;
  padding: 8px;
  text-transform: uppercase;
  border-radius: 5px;
  outline: none;
  width: 100%;
}

.signup-btn h3 {
  margin: 0px;
}

.signup-btn:hover {
  background-color: #48E0EA;
  color: white;
  cursor: pointer;
}

.signup-goback {
  display: flex;
  flex-direction: column;
  width: 100%;
}

.signup-goback h5 {
  margin: 10px;
  align-self: center;
}