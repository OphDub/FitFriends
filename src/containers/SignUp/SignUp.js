import React, { Component } from 'react';
import './SignUp.css';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { signup } from '../../thunks/thunks';

export class SignUp extends Component {
  constructor () {
    super();
    this.state={
      userEmail: '',
      userPass1: '',
      userPass2: '',
      errorMsg: ''
    };
  }

  handleChange = (e) => {
    const { name, value } = e.target;
    this.setState({ [name]: value });
  }

  signUpValidation = () => {
    const { userEmail, userPass1, userPass2 } = this.state;

    if (userEmail === '' || userPass1 === '' || userPass2 === '') {
      const errorMsg = 'Please fill out all fields.';

      this.setState({ errorMsg });
      return false;
    }

    if (userPass1 !== userPass2) {
      const errorMsg = 'Please make sure passwords match.';

      this.setState({ errorMsg });
      return false;
    }

    return true;
  }

  handleSignUp = async (e) => {
    e.preventDefault();
    const {userEmail, userPass1} = this.state;

    if (!this.signUpValidation()) {
      return;
    }

    try {
      await this.props.signup(userEmail, userPass1);

      this.setState({
        userEmail: '',
        userPass1: '',
        userPass2: '',
        errorMsg: ''
      });

      this.props.history.push('/workout');
    } catch (error) {
      const errorMsg = error.message;

      this.setState({ errorMsg });
      throw error;
    }
  }

  renderForm = () => {
    return (
      <form className="SignUp-form">
        <label htmlFor="userEmail">
          <span>Email: </span>
          <input type="email"
            placeholder="Email"
            name="userEmail"
            value={this.state.userEmail}
            onChange={this.handleChange}
          />
        </label>
        <label htmlFor="userPass1">
          <span>Password: </span>
          <input type="password"
            placeholder="Password"
            name="userPass1"
            value={this.state.userPass1}
            onChange={this.handleChange}
          />
        </label>
        <label htmlFor="userPass2">
          <span>Confirm Password: </span>
          <input type="password"
            placeholder="Password"
            name="userPass2"
            value={this.state.userPass2}
            onChange={this.handleChange}
          />
        </label>
        <button onClick={this.handleSignUp}
          className="signup-btn">
          <h3>Sign Up</h3>
        </button>
      </form>
    );
  }

  renderError = () => {
    return (
      <div className="signup-error">
        <h5 className="signup-error-msg">{this.state.errorMsg}</h5>
      </div>
    );
  }

  render () {
    return (
      <div className="SignUp">
        <div className="form-background">
          {this.renderForm()}
          {this.renderError()}
          <div className="signup-goback">
            <h5>Already have an account?</h5>
            <NavLink to="/login">
              <button className="signup-btn">
                Go to Login
              </button>
            </NavLink>
          </div>
        </div>
      </div>
    );
  }
}

SignUp.propTypes = {
  signup: PropTypes.func.isRequired
};

export const mapDispatchToProps = (dispatch) => ({
  signup: (email, password) => dispatch(signup(email, password))
});

export default withRouter(connect(null, mapDispatchToProps)(SignUp));
