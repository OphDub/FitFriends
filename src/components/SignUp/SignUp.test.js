/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import { SignUp, mapDispatchToProps } from './SignUp';

describe('SIGNUP', () => {
  let renderedComponent;
  let mockSignUp;

  beforeEach(() => {
    mockSignUp = jest.fn();
    renderedComponent = shallow(<SignUp signup={mockSignUp}/>);
  });

  it('should match snapshot', () => {
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should change state when handleChange is called', () => {
    const mockEvent = { target: { value: 'Big-O', name: 'userName'}};
    const expected = 'Big-O';

    renderedComponent.instance().handleChange(mockEvent);
    renderedComponent.update();

    expect(renderedComponent.state().userName).toEqual(expected);
  });

  it('should return false and set an error message in state if all user items in state are not complete', () => {
    renderedComponent.instance().setState({
      userEmail: 'will@me.com',
      userPass1: 'password',
      userPass2: '',
    });
    const expected = false;
    const expectedErrorMsg = 'Please fill out all fields.';

    expect(renderedComponent.instance().signUpValidation()).toEqual(expected);
    expect(renderedComponent.state().errorMsg).toEqual(expectedErrorMsg)
  });

  it('should return false and set an error message in state if passwords do not match', () => {
    renderedComponent.instance().setState({
      userEmail: 'will@me.com',
      userPass1: 'password',
      userPass2: 'different',
    });

    const expected = false;
    const expectedErrorMsg = 'Please make sure passwords match.';

    expect(renderedComponent.instance().signUpValidation()).toEqual(expected);
    expect(renderedComponent.state().errorMsg).toEqual(expectedErrorMsg)
  });

  it('should return true if all user elements in state are ok', () => {
    renderedComponent.instance().setState({
      userEmail: 'will@iambest.com',
      userPass1: 'password',
      userPass2: 'password',
    });
    const expected = true;

    expect(renderedComponent.instance().signUpValidation()).toEqual(expected);
  });

  it('should call signup when handleSignUp is called', () => {
    const mockEvent = { preventDefault: jest.fn() };

    renderedComponent.setState({
      userEmail: 'will@whodat.com',
      userPass1: 'password',
      userPass2: 'password',
    });
    renderedComponent.instance().handleSignUp(mockEvent);

    expect(mockSignUp).toHaveBeenCalled();
  });

  it('should clear state when handleSignUp is called', async () => {
    const mockEvent = { preventDefault: jest.fn() };
    const expected = {
      userEmail: '',
      userPass1: '',
      userPass2: '',
      errorMsg: '',
    };

    renderedComponent.setState({
      userEmail: 'will@me.com',
      userPass1: 'password',
      userPass2: 'password',
    });

    await renderedComponent.instance().handleSignUp(mockEvent);

    expect(renderedComponent.state()).toEqual(expected);
  });

  it('should call the dispatch fun when the signup func is called', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);

    mapped.signup();

    expect(mockDispatch).toHaveBeenCalled();
  });
})
