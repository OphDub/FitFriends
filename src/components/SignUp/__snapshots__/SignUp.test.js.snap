// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SIGNUP should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUp
    signup={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="form-background"
>
        <form
                className="SignUp-form"
        >
                <label
                        htmlFor="userEmail"
                >
                        <span>
                                Email: 
                        </span>
                        <input
                                name="userEmail"
                                onChange={[Function]}
                                placeholder="Email"
                                type="email"
                                value=""
                        />
                </label>
                <label
                        htmlFor="userPass1"
                >
                        <span>
                                Password: 
                        </span>
                        <input
                                name="userPass1"
                                onChange={[Function]}
                                placeholder="Password"
                                type="password"
                                value=""
                        />
                </label>
                <label
                        htmlFor="userPass2"
                >
                        <span>
                                Confirm Password: 
                        </span>
                        <input
                                name="userPass2"
                                onChange={[Function]}
                                placeholder="Password"
                                type="password"
                                value=""
                        />
                </label>
                <button
                        className="signup-btn"
                        onClick={[Function]}
                >
                        <h3>
                                Sign Up
                        </h3>
                </button>
        </form>
        <div
                className="signup-error"
        >
                <h5
                        className="signup-error-msg"
                />
        </div>
        <div
                className="signup-goback"
        >
                <h5>
                        Already have an account?
                </h5>
                <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        to="/login"
                >
                        <button
                                className="signup-btn"
                        >
                                Go to Login
                        </button>
                </NavLink>
        </div>
</div>,
      "className": "SignUp",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            className="SignUp-form"
>
            <label
                        htmlFor="userEmail"
            >
                        <span>
                                    Email: 
                        </span>
                        <input
                                    name="userEmail"
                                    onChange={[Function]}
                                    placeholder="Email"
                                    type="email"
                                    value=""
                        />
            </label>
            <label
                        htmlFor="userPass1"
            >
                        <span>
                                    Password: 
                        </span>
                        <input
                                    name="userPass1"
                                    onChange={[Function]}
                                    placeholder="Password"
                                    type="password"
                                    value=""
                        />
            </label>
            <label
                        htmlFor="userPass2"
            >
                        <span>
                                    Confirm Password: 
                        </span>
                        <input
                                    name="userPass2"
                                    onChange={[Function]}
                                    placeholder="Password"
                                    type="password"
                                    value=""
                        />
            </label>
            <button
                        className="signup-btn"
                        onClick={[Function]}
            >
                        <h3>
                                    Sign Up
                        </h3>
            </button>
</form>,
          <div
            className="signup-error"
>
            <h5
                        className="signup-error-msg"
            />
</div>,
          <div
            className="signup-goback"
>
            <h5>
                        Already have an account?
            </h5>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        to="/login"
            >
                        <button
                                    className="signup-btn"
                        >
                                    Go to Login
                        </button>
            </NavLink>
</div>,
        ],
        "className": "form-background",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                htmlFor="userEmail"
>
                <span>
                                Email: 
                </span>
                <input
                                name="userEmail"
                                onChange={[Function]}
                                placeholder="Email"
                                type="email"
                                value=""
                />
</label>,
              <label
                htmlFor="userPass1"
>
                <span>
                                Password: 
                </span>
                <input
                                name="userPass1"
                                onChange={[Function]}
                                placeholder="Password"
                                type="password"
                                value=""
                />
</label>,
              <label
                htmlFor="userPass2"
>
                <span>
                                Confirm Password: 
                </span>
                <input
                                name="userPass2"
                                onChange={[Function]}
                                placeholder="Password"
                                type="password"
                                value=""
                />
</label>,
              <button
                className="signup-btn"
                onClick={[Function]}
>
                <h3>
                                Sign Up
                </h3>
</button>,
            ],
            "className": "SignUp-form",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    Email: 
</span>,
                  <input
                    name="userEmail"
                    onChange={[Function]}
                    placeholder="Email"
                    type="email"
                    value=""
/>,
                ],
                "htmlFor": "userEmail",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Email: ",
                  },
                  "ref": null,
                  "rendered": "Email: ",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "name": "userEmail",
                    "onChange": [Function],
                    "placeholder": "Email",
                    "type": "email",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    Password: 
</span>,
                  <input
                    name="userPass1"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
/>,
                ],
                "htmlFor": "userPass1",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Password: ",
                  },
                  "ref": null,
                  "rendered": "Password: ",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "name": "userPass1",
                    "onChange": [Function],
                    "placeholder": "Password",
                    "type": "password",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    Confirm Password: 
</span>,
                  <input
                    name="userPass2"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
/>,
                ],
                "htmlFor": "userPass2",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Confirm Password: ",
                  },
                  "ref": null,
                  "rendered": "Confirm Password: ",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "name": "userPass2",
                    "onChange": [Function],
                    "placeholder": "Password",
                    "type": "password",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h3>
                  Sign Up
</h3>,
                "className": "signup-btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sign Up",
                },
                "ref": null,
                "rendered": "Sign Up",
                "type": "h3",
              },
              "type": "button",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h5
              className="signup-error-msg"
/>,
            "className": "signup-error",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "signup-error-msg",
            },
            "ref": null,
            "rendered": "",
            "type": "h5",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h5>
                Already have an account?
</h5>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/login"
>
                <button
                                className="signup-btn"
                >
                                Go to Login
                </button>
</NavLink>,
            ],
            "className": "signup-goback",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Already have an account?",
              },
              "ref": null,
              "rendered": "Already have an account?",
              "type": "h5",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": <button
                  className="signup-btn"
>
                  Go to Login
</button>,
                "to": "/login",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Go to Login",
                  "className": "signup-btn",
                },
                "ref": null,
                "rendered": "Go to Login",
                "type": "button",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="form-background"
>
          <form
                    className="SignUp-form"
          >
                    <label
                              htmlFor="userEmail"
                    >
                              <span>
                                        Email: 
                              </span>
                              <input
                                        name="userEmail"
                                        onChange={[Function]}
                                        placeholder="Email"
                                        type="email"
                                        value=""
                              />
                    </label>
                    <label
                              htmlFor="userPass1"
                    >
                              <span>
                                        Password: 
                              </span>
                              <input
                                        name="userPass1"
                                        onChange={[Function]}
                                        placeholder="Password"
                                        type="password"
                                        value=""
                              />
                    </label>
                    <label
                              htmlFor="userPass2"
                    >
                              <span>
                                        Confirm Password: 
                              </span>
                              <input
                                        name="userPass2"
                                        onChange={[Function]}
                                        placeholder="Password"
                                        type="password"
                                        value=""
                              />
                    </label>
                    <button
                              className="signup-btn"
                              onClick={[Function]}
                    >
                              <h3>
                                        Sign Up
                              </h3>
                    </button>
          </form>
          <div
                    className="signup-error"
          >
                    <h5
                              className="signup-error-msg"
                    />
          </div>
          <div
                    className="signup-goback"
          >
                    <h5>
                              Already have an account?
                    </h5>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              to="/login"
                    >
                              <button
                                        className="signup-btn"
                              >
                                        Go to Login
                              </button>
                    </NavLink>
          </div>
</div>,
        "className": "SignUp",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <form
              className="SignUp-form"
>
              <label
                            htmlFor="userEmail"
              >
                            <span>
                                          Email: 
                            </span>
                            <input
                                          name="userEmail"
                                          onChange={[Function]}
                                          placeholder="Email"
                                          type="email"
                                          value=""
                            />
              </label>
              <label
                            htmlFor="userPass1"
              >
                            <span>
                                          Password: 
                            </span>
                            <input
                                          name="userPass1"
                                          onChange={[Function]}
                                          placeholder="Password"
                                          type="password"
                                          value=""
                            />
              </label>
              <label
                            htmlFor="userPass2"
              >
                            <span>
                                          Confirm Password: 
                            </span>
                            <input
                                          name="userPass2"
                                          onChange={[Function]}
                                          placeholder="Password"
                                          type="password"
                                          value=""
                            />
              </label>
              <button
                            className="signup-btn"
                            onClick={[Function]}
              >
                            <h3>
                                          Sign Up
                            </h3>
              </button>
</form>,
            <div
              className="signup-error"
>
              <h5
                            className="signup-error-msg"
              />
</div>,
            <div
              className="signup-goback"
>
              <h5>
                            Already have an account?
              </h5>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            to="/login"
              >
                            <button
                                          className="signup-btn"
                            >
                                          Go to Login
                            </button>
              </NavLink>
</div>,
          ],
          "className": "form-background",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label
                  htmlFor="userEmail"
>
                  <span>
                                    Email: 
                  </span>
                  <input
                                    name="userEmail"
                                    onChange={[Function]}
                                    placeholder="Email"
                                    type="email"
                                    value=""
                  />
</label>,
                <label
                  htmlFor="userPass1"
>
                  <span>
                                    Password: 
                  </span>
                  <input
                                    name="userPass1"
                                    onChange={[Function]}
                                    placeholder="Password"
                                    type="password"
                                    value=""
                  />
</label>,
                <label
                  htmlFor="userPass2"
>
                  <span>
                                    Confirm Password: 
                  </span>
                  <input
                                    name="userPass2"
                                    onChange={[Function]}
                                    placeholder="Password"
                                    type="password"
                                    value=""
                  />
</label>,
                <button
                  className="signup-btn"
                  onClick={[Function]}
>
                  <h3>
                                    Sign Up
                  </h3>
</button>,
              ],
              "className": "SignUp-form",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span>
                      Email: 
</span>,
                    <input
                      name="userEmail"
                      onChange={[Function]}
                      placeholder="Email"
                      type="email"
                      value=""
/>,
                  ],
                  "htmlFor": "userEmail",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Email: ",
                    },
                    "ref": null,
                    "rendered": "Email: ",
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "name": "userEmail",
                      "onChange": [Function],
                      "placeholder": "Email",
                      "type": "email",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                ],
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span>
                      Password: 
</span>,
                    <input
                      name="userPass1"
                      onChange={[Function]}
                      placeholder="Password"
                      type="password"
                      value=""
/>,
                  ],
                  "htmlFor": "userPass1",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Password: ",
                    },
                    "ref": null,
                    "rendered": "Password: ",
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "name": "userPass1",
                      "onChange": [Function],
                      "placeholder": "Password",
                      "type": "password",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                ],
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span>
                      Confirm Password: 
</span>,
                    <input
                      name="userPass2"
                      onChange={[Function]}
                      placeholder="Password"
                      type="password"
                      value=""
/>,
                  ],
                  "htmlFor": "userPass2",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Confirm Password: ",
                    },
                    "ref": null,
                    "rendered": "Confirm Password: ",
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "name": "userPass2",
                      "onChange": [Function],
                      "placeholder": "Password",
                      "type": "password",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                ],
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <h3>
                    Sign Up
</h3>,
                  "className": "signup-btn",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Sign Up",
                  },
                  "ref": null,
                  "rendered": "Sign Up",
                  "type": "h3",
                },
                "type": "button",
              },
            ],
            "type": "form",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h5
                className="signup-error-msg"
/>,
              "className": "signup-error",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "signup-error-msg",
              },
              "ref": null,
              "rendered": "",
              "type": "h5",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h5>
                  Already have an account?
</h5>,
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  to="/login"
>
                  <button
                                    className="signup-btn"
                  >
                                    Go to Login
                  </button>
</NavLink>,
              ],
              "className": "signup-goback",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Already have an account?",
                },
                "ref": null,
                "rendered": "Already have an account?",
                "type": "h5",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": <button
                    className="signup-btn"
>
                    Go to Login
</button>,
                  "to": "/login",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Go to Login",
                    "className": "signup-btn",
                  },
                  "ref": null,
                  "rendered": "Go to Login",
                  "type": "button",
                },
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
